covid_moonshot/analysis/structures.py

```diff
@@ -119,6 +119,118 @@ def mdtraj_to_oemol(snapshot: md.Trajectory):
                 return mol
 
 
+def cluster_snaphots(
+    project_path: str,
+    project_data_path: str,
+    run: int,
+    frame: int,
+    fragment_id: str,
+    n_snapshots: Optional[int],
+    cache_dir: Optional[str],
+):
+    """
+    Extract the specified snapshot, align it to the reference fragment, and write protein and ligands to separate PDB files
+
+    Parameters
+    ----------
+    project_path : str
+       Path to project directory (e.g. '/home/server/server2/projects/13422')
+    run : str or int
+       Run (e.g. '0')
+    frame : int
+    fragment_id : str
+      Fragment ID (e.g. 'x10789')
+    n_snapshots : int, default=20
+        number of snapshots to load for simulation
+    cache_dir : str or None
+       If specified, cache relevant parts of "htf.npz" file in a local directory of this name
+
+    Returns
+    -------
+    sliced_snapshot : dict of str : mdtraj.Trajectory
+      sliced_snapshot[name] is the Trajectory for name in ['protein', 'old_ligand', 'new_ligand', 'old_complex', 'new_complex']
+    components : dict of str : oechem.OEMol
+      components[name] is the OEMol for name in ['protein', 'old_ligand', 'new_ligand']
+
+    """
+    import random
+    import numpy as np
+    from scipy.cluster.hierarchy import ward, fcluster
+    from scipy.spatial.distance import pdist
+
+    # open n random snaphots for the RUN
+    for i in range(0, n_snapshots):
+        # TODO un-hardcode these
+        clone = random.randint(0,99)
+        gen = random.randint(0,2)
+        if i == 0:
+            # TODO safeguard against trying to load output that doesn't exist --- chose n_snapshots randomly from those that exists on disk
```

Comment:
Passing just a list of clones/gens sounds reasonable to me (especially if we're not actually going to use the works here). The `list_results` function in `lib.py` gets a listing of clones and gens given a project path and run.

Actually, it might make sense to change things around a little bit to call `list_path` just once in `analyze_run`, and pass the result to both `extract_works` and to this function. (We can always clean this up later, too)